container-toolkit/cdi-support.md

```diff
@@ -29,7 +30,68 @@ CDI also improves the compatibility of the NVIDIA container stack with certain f
 
 - You installed an NVIDIA GPU Driver.
 
-### Procedure
+### Automatic CDI Specification Generation (v1.18.0+)
+
+As of NVIDIA Container Toolkit `v1.18.0`, the CDI specification is automatically generated and updated by a systemd service called `nvidia-cdi-refresh`. This service:
+
+- Automatically generates the CDI specification at `/var/run/cdi/nvidia.yaml` when NVIDIA drivers are installed or upgraded
+- Monitors changes to driver-related files (`modules.dep` and `modules.dep.bin`) to trigger regeneration
+- Runs automatically on system boot to ensure the specification is up to date
+- Is enabled and started automatically during fresh installations
+
+The automatic service handles the following scenarios:
+- First-time driver installation
+- Driver upgrades
+- System reboots
+
+```{note}
+The automatic CDI refresh service does not handle:
+- Driver removal (the CDI file is intentionally preserved)
+- Runtime topology changes (MIG, hot-plug, module unload/load)
+- Manual configuration changes
+
+For these scenarios, you may still need to manually regenerate the CDI specification.
+```
+
+#### Customizing the Automatic CDI Refresh Service
+
+You can customize the behavior of the `nvidia-cdi-refresh` service by adding environment variables to `/etc/nvidia-container-toolkit/cdi-refresh.env`. This file is read by the service and allows you to modify the `nvidia-ctk cdi generate` command behavior.
+
+Example configuration file:
+```bash
+# /etc/nvidia-container-toolkit/cdi-refresh.env
+NVIDIA_CTK_DEBUG=1
+# Add other nvidia-ctk environment variables as needed
+```
+
+#### Managing the CDI Refresh Service
+
+The `nvidia-cdi-refresh` service consists of two systemd units:
+
+- `nvidia-cdi-refresh.path` - Monitors for changes to driver files and triggers the service
+- `nvidia-cdi-refresh.service` - Executes the CDI specification generation
+
+You can manage these services using standard systemd commands:
+
+```console
+# Check service status
+$ sudo systemctl status nvidia-cdi-refresh.path
+$ sudo systemctl status nvidia-cdi-refresh.service
+
+# Manually trigger CDI generation
+$ sudo systemctl start nvidia-cdi-refresh.service
+
+# Enable/disable the automatic monitoring
+$ sudo systemctl enable nvidia-cdi-refresh.path
+$ sudo systemctl disable nvidia-cdi-refresh.path
+
+# View service logs
+$ sudo journalctl -u nvidia-cdi-refresh.service
+```
+
+### Manual CDI Specification Generation
```

Comment:
Given the automatic process now writes to `/var/run/cdi/nvidia.yaml` we should update the instructions below to output to this file instead.

Reply:
done